worktree.go

```diff
@@ -997,6 +1009,10 @@ func (gr GrepResult) String() string {
 
 // Grep performs grep on a repository.
 func (r *Repository) Grep(opts *GrepOptions) ([]GrepResult, error) {
+	if opts == nil {
+		opts = &GrepOptions{}
```

Comment:
We should exit with an error here, as per [this](https://github.com/go-git/go-git/issues/1051#issuecomment-2443482527).